src/evict/evict_page.c

```diff
@@ -179,9 +179,14 @@ __evict_delete_ref(WT_SESSION_IMPL *session, WT_REF *ref, bool closing)
 		 * something is busy, be sure that the page still ends up
 		 * marked deleted.
 		 */
-		if (ndeleted > pindex->entries / 10 && pindex->entries > 1 &&
-		    (ret = __wt_split_reverse(session, ref)) != EBUSY)
-			return (ret);
+		if (ndeleted > pindex->entries / 10 && pindex->entries > 1) {
+			if ((ret = __wt_split_reverse(session, ref)) == 0)
+				return (ret);
+			WT_RET_BUSY_OK(ret);
```

Comment:
@keithbostic I don't see why this is necessary. If the reverse split failed with EBUSY, we still want to switch the ref back to ``WT_REF_DELETED``, don't we? Is it possible that we never switched it to locked - in that case I'd be inclined to update the ``WT_ASSERT`` to be ``WT_ASSERT(session, ret == EBUSY || ref->state == WT_REF_LOCKED)``, and let ``EBUSY`` fall through (see following).